File: src/ios/AppChange.m
#import "AppChange.h"
#import <Cordova/CDV.h>

@implementation AppChange
- (void)check:(CDVInvokedUrlCommand*)command {
    
    CDVPluginResult* pluginResult = nil;
    
    NSString* scheme = [command.arguments objectAtIndex:0];
	    
    if ([[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:scheme]]) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:(true)];
    }
    else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsBool:(false)];
    }
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}

- (void)inport:(uri)command{
	
}

- (void)toAPPopen:()command{
	CDVPluginResult* pluginResult = nil;
    NSString* urischeme = [command.arguments objectAtIndex:0];
	if ([[UIApplication sharedApplication]canOpenURL:[NSURL URLWithString:@urischeme]]) {
		[[UIApplication sharedApplication]openURL:[NSURL URLWithString:@urischeme]];
	}
}

- (void)goAPPurl:()command{

	CDVPluginResult* pluginResult = nil;
    NSString* appid = [command.arguments objectAtIndex:0];
	NSString *str = [NSString stringWithFormat:@"http://itunes.apple.com/app/id%d",appid];
	[[UIApplication sharedApplication] openURL:[NSURL urlWithString:str]];	
}
@end

File: src/ios/AppChange.swift
import Foundation

 @objc(HWPAppChange) class AppChange : CDVPlugin {
    func check(command: CDVInvokedUrlCommand) {
       let id = command.arguments[0] as? String ?? ""
        let URLs = command.arguments[1] as? String ?? ""
        let callsheme:NSURL = NSURL(string: id)!
            if(UIApplication.sharedApplication().canOpenURL(callsheme)){
                let pluginYES = CDVPluginResult(status: CDVCommandStatus_OK, messageAsString: id)
                commandDelegate!.sendPluginResult(pluginYES, callbackId:command.callbackId)
            }else{
                let pluginNO = CDVPluginResult(status: CDVCommandStatus_ERROR, messageAsString: URLs)
                commandDelegate!.sendPluginResult(pluginNO, callbackId:command.callbackId)
            }
    }
    func goAPPurl(command: CDVInvokedUrlCommand) {
        let id = command.arguments[0] as? String ?? ""
        if(id != ""){
            let urls:NSURL = NSURL(string: "itms-apps://itunes.apple.com/app/id\(id)")!
            UIApplication.sharedApplication().openURL(urls)
        }else{
            let pluginNO = CDVPluginResult(status: CDVCommandStatus_ERROR, messageAsString: "INSTALL APP ERROR!!")
            commandDelegate!.sendPluginResult(pluginNO , callbackId: command.callbackId)
        }
    }
    func toAPPopen(command: CDVInvokedUrlCommand) {
        let sheme = command.arguments[0] as? String ?? "OPEN APP ERROR!!"
        let uris:NSURL = NSURL(string: sheme)!
        if(UIApplication.sharedApplication().canOpenURL(uris)){
            UIApplication.sharedApplication().openURL(uris)
        }else{
            let pluginNO = CDVPluginResult(status: CDVCommandStatus_ERROR, messageAsString: "OPEN APP ERROR!!")
            commandDelegate!.sendPluginResult(pluginNO , callbackId: command.callbackId)
        }
    }
    func application(application: UIApplication, handleOpenURL url: NSURL) -> Bool {
        return true;
    }

}
